purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1837,12 +1859,36 @@ class Purchases internal constructor(
                         )
                         deviceCache.addSuccessfullyPostedToken(purchaseToken)
                     }
-                    onError(error)
+                    if (offlineEntitlementsManager.shouldCalculateOfflineCustomerInfoInPostReceipt(isServerError)) {
+                        calculateOfflineCustomerInfo(appUserID, { onSuccess() }, { onError(it) })
+                    } else {
+                        onError(error)
+                    }
```

Comment:
There is a lot of duplication in this file... We post the receipt in 3 different places (purchases, sync purchases and restore purchases), and we need to handle all 3. I first tried to extract the post receipt flow to a different class so we could reuse more code, but it got tricky since they aren't exactly the same... I decided to delay that refactor for now, in favor of getting this done quickly.

Reply:
makes sense. It feels like at the very least we can turn these into one-liners and reuse that (as a follow-up)

Reply:
In the end, I worked on that refactor in #967 ... I think it helps to separate that logic and not have it in `Purchases`, specially when writing the tests. It was a big refactor though...